AbstractDesignPatterns/Program.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+
+namespace AbstractDesignPatterns
+{
+
+    class Program
+    {
+        static void Main(string[] args) // client 
+        {
+
+            IPhoneProduct appleFactory = new AppleFactory();
+            appleFactory.CreatePhone();
+        
+
+            IPhoneProduct samsungFactory = new SamsungFactory();
```

Comment:
You have declared `samsungFactory ` using `IPhoneProduct` it is like having a factory of type product! instead it should be like this 
```c#
IPhoneProduct samsungFactory = new SamsungFactory();
```